sdk/azcore/core.go

```diff
@@ -109,3 +110,38 @@ func (n nopCloser) Close() error {
 func NopCloser(rs io.ReadSeeker) ReadSeekCloser {
 	return nopCloser{rs}
 }
+
+// Poller provides operations for checking the state of a long-running operation.
+// An LRO can be in either a non-terminal or terminal state.  A non-terminal state
+// indicates the LRO is still in progress.  A terminal state indicates the LRO has
+// completed successfully, failed, or was cancelled.
+type Poller interface {
+	// Done returns true if the LRO has reached a terminal state.
+	Done() bool
+
+	// Poll fetches the latest state of the LRO.  It returns an HTTP response or error.
+	// If the LRO has completed successfully, the poller's state is update and the HTTP
+	// response is returned.
+	// If the LRO has completed with failure or was cancelled, the poller's state is
+	// updated and the error is returned.
+	// If the LRO has not reached a terminal state, the poller's state is updated and
+	// the latest HTTP response is returned.
+	// If Poll fails, the poller's state is unmodified and the error is returned.
+	// Calling Poll on an LRO that has reached a terminal state will return the final
+	// HTTP response or error.
```

Comment:
Calling Poll may sometimes return the final response, but not always. There are several cases where FinalResponse needs to be called to get the true final response.

Reply:
It should provide the final HTTP response of the LRO (terminal/non-terminal state).  You are correct though that it might not contain the final result (i.e. the response body of the modified object).